### DISCUSSION TITLE
Is SHCOARSE (in example) assumed to be the beginning of the user data field?

### DISCUSSION THREAD

**Discussion by haisamido** [CCSDSPy/ccsdspy#109] [Q&A]
I'm fairly new to this module so pardon my rudimentary question. In the below [example](https://github.com/CCSDSPy/ccsdspy?tab=readme-ov-file#usage-example), is SHCOARSE assumed to be the beginning of the user data field, i.e. after primary and secondary headers?

```python
pkt = ccsdspy.FixedLength([
  PacketField(name='SHCOARSE', data_type='uint', bit_length=32),
  PacketField(name='SHFINE',   data_type='uint', bit_length=20),
  PacketField(name='OPMODE',   data_type='uint', bit_length=3),
  PacketField(name='SPACER',   data_type='fill', bit_length=1),
  PacketField(name='VOLTAGE',  data_type='int',  bit_length=8),
....
])
```

**Comment by ddasilva** [comment]
No problem @haisamido, long time no chat (I think we chatted on the NASA slack a few times a long time ago). 
In this example SHCOARSE is the first field to parse after the primary header. Let me know if that answers your question.

**Reply by haisamido** [reply]
Thank you for your answer. Long time indeed @ddasilva :-). What if one wanted to also capture header metadata as separate telemetered parameters, such as:

```python
  PacketField(name='packet_version_number', data_type='uint', bit_length=3 ),
...
  PacketField(name='packet_data_length', data_type='uint', bit_length=16),
```
I ask this because this is a nominal feature, where I work, where one can query the `packet_version_number`, and any of the header metadata, such as, `packet_type`, `secondary_header_flag`, etc.

**Reply by ddasilva** [reply]
Of course! With `FixedLength.load()` and `VariableLength.load()`, you can pass a keyword argument called `include_primary_header=True` to get the primary header fields back in the result dictionary.

The names of the primary header fields it adds to the dictionary are:
`CCSDS_VERSION_NUMBER`, `CCSDS_PACKET_TYPE`, `CCSDS_SECONDARY_FLAG`, `CCSDS_APID`, `CCSDS_SEQUENCE_FLAG`, `CCSDS_SEQUENCE_COUNT`, and `CCSDS_PACKET_LENGTH`

There is also a shortcut function [ccsdspy.utils.read_primary_headers(file)](https://docs.ccsdspy.org/en/latest/api/ccsdspy.utils.read_primary_headers.html#ccsdspy.utils.read_primary_headers) if you don't care about also parsing the body of the packet.

**Reply by haisamido** [reply]
Excellent! Thank you.
